### DISCUSSION TITLE
Old plugin releases get deleted

### DISCUSSION THREAD

**Discussion by averma21** [alexklibisz/elastiknn#371] [Q&A] [answered]
A few months back I downloaded 7.9.x.x version of elastiknn plugin but now I can't find them in the releases page - https://github.com/alexklibisz/elastiknn/tags

I see that you have mentioned - "For now I’m only releasing the plugin for a single Elasticsearch version. I’d like to eventually come back and implement releases for multiple versions." here - https://elastiknn.com/installation/

The problem with this is that if I don't upgrade to latest version of elasticsearch everytime, I would myself have to keep the older plugin release downloaded somewhere.

Could you please elaborate the reason behind deletion of older versions of the plugin? Would it not make sense to just keep them available for downloads?

**Comment by alexklibisz** [comment, marked as answer]
Hi, they don't actually get deleted. Perhaps it's a bit confusing because I changed the version naming from `0.1.0-pre<N>` to `<es version>.<N>` about a month ago.

Here is the latest release for 7.9.3: https://github.com/alexklibisz/elastiknn/releases/tag/0.1.0-PRE50

I do not plan to support multiple versions at once. I tried setting this up for several hours and couldn't figure it out. The ES folks also introduce breaking internal changes between Elasticsearch versions, so it's not as simple as just iterating over a list of dependency versions. The only way to do it would be to keep around several branches or several copy/pasted directories of the source code, and both of those are not worth the time IMO. The project is pretty stable now, so I'll stay on top of the official Elasticsearch releases as they come out.
